caddyhttp/fastcgi/fastcgi.go

```diff
@@ -73,9 +82,33 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 
 			// Connect to FastCGI gateway
 			network, address := rule.parseAddress()
-			fcgiBackend, err := Dial(network, address)
-			if err != nil {
-				return http.StatusBadGateway, err
+
+			var fcgiBackend *FCGIClient
+			var mut *sync.Mutex
+			ok := false
+			if usePersistentFcgiConnections {
+				// re use connection, if possible
+				if persistentConnections == nil {
+					persistentConnections = make(map[string]*FCGIClient)
+					poorMansSerialisation = make(map[string]*sync.Mutex)
+				}
+				mut, ok = poorMansSerialisation[network+address]
+				if !ok || mut == nil {
+					poorMansSerialisation[network+address] = new(sync.Mutex)
+				}
+				poorMansSerialisation[network+address].Lock()
+				defer poorMansSerialisation[network+address].Unlock()
+
+				fcgiBackend, ok = persistentConnections[network+address]
+			}
+
+			// otherwise dial:
+			if !ok || fcgiBackend == nil {
```

Comment:
Same here, don't need the second check if `ok` entries always have a non-nil value.